src/drawing/path.js

```diff
@@ -24,6 +24,7 @@ export default function(context, options = {}, fn) {
 
   context.beginPath();
   context.strokeStyle = color || context.strokeStyle;
+
   context.lineWidth =
     lineWidth ||
     (lineWidth === undefined && toolStyle.getToolWidth()) ||
```

Comment:
I know this isn't your code, but this logic looks wrong. Why would we set it to true?

Reply:
> Why would we set it to true?

If you mean `context.lineWidth = lineWidth || ...`, that's just checking if its truthy, i.e. set and not zero.

Reply:
`lineWidth` comes in as a parameter. According to JSDoc, it can be a `number` or `undefined`.


```js
// System under test
 context.lineWidth =
    lineWidth ||
    (lineWidth === undefined && toolStyle.getToolWidth()) ||
    context.lineWidth;

// Input --> Output
// 1 --> 1 (set to provided)
// 0 --> context.lineWidth (use current)
// undefined --> true (if toolStyle.getToolWidth() returns non-zero number)
// undefined --> context.lineWidth (if toolStyle.getToolWidth() returns 0)
```

I _think_ that's right, but that means in some cases we could accidentally set `context.lineWidth` to `true`

Reply:
The JSdoc specifies that it can be `null` or `undefined`:

> The width of lines in the path. If null, no line width is set. If undefined then toolStyle.getToolWidth() is set.

^ I think the wording is maybe a bit off, either way we need a way to not set the line, and null is probably the best way? undefined means you haven't set it, and some default should be use.

> I think that's right, but that means in some cases we could accidentally set context.lineWidth to true.

If `lineWidth === undefined`, `(lineWidth === undefined && toolStyle.getToolWidth())` evalutes to the valye of `toolStyle.getToolWidth()`, not `true`, e.g.:

```
const lineWidth = undefined;
function getLineWidth () {return 10;}
const lw == lineWidth || (lineWidth === undefined && getLineWidth());

// lw = 10.
```